components/ws-manager/pkg/manager/metrics.go

```diff
@@ -242,19 +244,39 @@ func (m *metrics) Register(reg prometheus.Registerer) error {
 }
 
 func (m *metrics) OnChange(status *api.WorkspaceStatus) {
+	clog := log.WithFields(log.OWI(status.Metadata.Owner, status.Metadata.MetaId, status.Id))
+
 	var removeFromState bool
 	tpe := api.WorkspaceType_name[int32(status.Spec.Type)]
 	m.mu.Lock()
 	defer func() {
 		if removeFromState {
 			delete(m.phaseState, status.Id)
-		} else {
+		} else if m.phaseState[status.Id] != status.Phase {
 			m.phaseState[status.Id] = status.Phase
+
+			// There are some conditions we'd like to get notified about, for example while running experiements or because
+			// they represent out-of-the-ordinary situations.
+			// We attempt to use the GCP Error Reporting for this, hence log these situations as errors.
+			if status.Conditions.Failed != "" {
```

Comment:
I think we would now miss a log if `removeFromState == true`, e.g. when the workspace status becomes `STOPPED`. Should this if check be a separate check within the `defer` function so that it always runs?

Reply:
Nice catch @WVerlaek
I did an update, PTAL!